### PR TITLE
feat: Add velero release 1.7.1

### LINKED ISSUE #2663: Restic support in Velero plugin
## Feature Request

Up to now, Velero plugin's Restic function support is still restricted. Only enable or disable option exists.
Need to add more Restic related options to support detailed Restic setting.

## Describe alternatives you've considered

No alternatives is coming to my mind yet.

## Additional context

 I am working on the Velero Carvel supporting now. I'll make a PR later.

### PR DESCRIPTION
**Pull request by blackpiglet**
1. port existing 1.6.3 function to 1.7.1
2. update upstream file to velero 1.7.1
3. add restic support
4. update repos

Signed-off-by: Xun Jiang <jxun@vmware.com>

## What this PR does / why we need it
Add Velero release v1.7.1. Add restic supporting function in v1.7.1

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
1. port existing 1.6.3 function to 1.7.1
2. add restic support in 1.7.1
```

## Which issue(s) this PR fixes

Fixes: #2663

## Describe testing done for PR
Create standalone cluster in Docker. 

GINKGO_FOCUS='Basic' GITHUB_TOKEN=$GITHUB_TOKEN REGISTRY_CREDENTIAL_FILE=/Users/jxun/.docker/config.json make e2e-test
GINKGO_FOCUS='Snapshot' GITHUB_TOKEN=$GITHUB_TOKEN REGISTRY_CREDENTIAL_FILE=/Users/jxun/.docker/config.json make e2e-test

## Special notes for your reviewer
None

### CODE REVIEW

**Review by seemiller — CHANGES_REQUESTED** [review]
Looks good!

I did push this up to the TCE Harbor project and deployed it to my cluster.

Can you update the package yaml to include this image:
```
projects.registry.vmware.com/tce/velero@sha256:fe89fb0897df4f9df2665c305909c6cc5cda42bacf2fbfc2e111764e0431baf8
```

**Comment by blackpiglet** [comment]
@seemiller 
Thanks, Nick! 
I updated the package image bundle. Please take another look when you have time.

**Comment by dvonthenen** [comment]
going to merge based on this comment (https://github.com/vmware-tanzu/community-edition/pull/2700#pullrequestreview-830760417)... should we need an update, we can do on a follow up PR

File: addons/packages/velero/1.7.1/package.yaml [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+              description: The name for the service account
+              default: velero
+            annotations:
+              type: object
+              additionalProperties: string
+              description: Annotations to set on the Velero service account.
+              default: {}
+            labels:
+              type: object
+              additionalProperties: array
```
Comment (seemiller) [inline addons/packages/velero/1.7.1/package.yaml:412]:
```suggestion
              additionalProperties:
                type: array
                items:
                  type: string
```

File: addons/packages/velero/1.7.1/package.yaml [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+          type: object
+          description: Information about the Kubernetes service account Velero uses.
+          properties:
+            name:
+              type: string
+              description: The name for the service account
+              default: velero
+            annotations:
+              type: object
+              additionalProperties: string
```
Comment (seemiller) [inline addons/packages/velero/1.7.1/package.yaml:407]:
```suggestion
              additionalProperties:
                type: string
```

**Review by seemiller — DISMISSED** [review]

File: addons/packages/velero/1.7.1/package.yaml (hunk 2 above)
Comment (blackpiglet) [reply addons/packages/velero/1.7.1/package.yaml:407]: Thanks. It's reasonable to use string map to represent the annotations.

File: addons/packages/velero/1.7.1/package.yaml (hunk 1 above)
Comment (blackpiglet) [reply addons/packages/velero/1.7.1/package.yaml:412]: Please correct me, if I am wrong. 
In this way, it use map[string]string to represent "labels"
  additionalProperties:
     type: string

By that way, it use map[[]string]string to represent "labels"
              additionalProperties:
                type: array
                items:
                  type: string

I think maybe using first option is more reasonable.

File: addons/packages/velero/metadata.yaml [hunk 3]
```diff
@@ -9,5 +9,6 @@ spec:
   providerName: VMware
   maintainers:
     - name: Carlisia Thompson
```
Comment (nrb) [inline addons/packages/velero/metadata.yaml:11]: We should probably remove Carlisia's name as part of this.

File: addons/packages/velero/1.7.1/package.yaml (hunk 1 above)
Comment (seemiller) [reply addons/packages/velero/1.7.1/package.yaml:413]: @blackpiglet If you feel that's correct, then make your change. Mine was just a suggestion.

File: addons/packages/velero/metadata.yaml (hunk 3 above)
Comment (blackpiglet) [reply addons/packages/velero/metadata.yaml:11]: OK.

**Comment by blackpiglet** [comment]
@nrb @seemiller 
Hi, guys. Could you help to take another look when you have time?

**Review by nrb — APPROVED** [review]
LGTM, but I'm less knowledgeable about ytt than Nick. If all this is doing is bumping the release version and copying it into the new directories, then I'm ok with merging.

**Review by seemiller — APPROVED** [review]

**Pull request merged**